app/uk/gov/hmrc/cardpaymentfrontend/models/extendedorigins/ExtendedPfVat.scala

```diff
@@ -17,30 +17,63 @@
 package uk.gov.hmrc.cardpaymentfrontend.models.extendedorigins
 
 import payapi.cardpaymentjourney.model.journey.{JourneySpecificData, JsdPfVat}
+import payapi.corcommon.model.taxes.other.XRef14Char
+import payapi.corcommon.model.taxes.vat.Vrn
 import play.api.mvc.AnyContent
 import uk.gov.hmrc.cardpaymentfrontend.actions.JourneyRequest
 import uk.gov.hmrc.cardpaymentfrontend.models.PaymentMethod._
 import uk.gov.hmrc.cardpaymentfrontend.models.openbanking.{OriginSpecificSessionData, PfVatSessionData}
 import uk.gov.hmrc.cardpaymentfrontend.models.{CheckYourAnswersRow, PaymentMethod}
 
 object ExtendedPfVat extends ExtendedOrigin {
-  override val serviceNameMessageKey: String = "add.message.key.here"
+  override val serviceNameMessageKey: String = "service-name.PfVat"
   override val taxNameMessageKey: String = "payment-complete.tax-name.PfVat"
-  def cardFeesPagePaymentMethods: Set[PaymentMethod] = Set.empty[PaymentMethod]
+  def cardFeesPagePaymentMethods: Set[PaymentMethod] = Set(OpenBanking, VariableDirectDebit)
   def paymentMethods(): Set[PaymentMethod] = Set(Card, OpenBanking, VariableDirectDebit, Bacs)
 
-  override def checkYourAnswersReferenceRow(journeyRequest: JourneyRequest[AnyContent])(payFrontendBaseUrl: String): Option[CheckYourAnswersRow] = None
+  private def vrn: JourneySpecificData => Option[Vrn] = {
+    case j: JsdPfVat => j.vrn
+    case _           => throw new RuntimeException("Incorrect origin found")
+  }
 
-  override def openBankingOriginSpecificSessionData: JourneySpecificData => Option[OriginSpecificSessionData] = {
-    case j: JsdPfVat => Some(PfVatSessionData(j.vrn, j.chargeRef))
+  override def checkYourAnswersReferenceRow(journeyRequest: JourneyRequest[AnyContent])(payFrontendBaseUrl: String): Option[CheckYourAnswersRow] = {
+    vrn(journeyRequest.journey.journeySpecificData).map { vrn =>
+      CheckYourAnswersRow(
+        titleMessageKey = "check-your-details.PfVat.reference",
+        value           = Seq(vrn.canonicalizedValue),
+        changeLink      = None
+      )
+    }
+  }
+
+  private def chargeReference: JourneySpecificData => Option[XRef14Char] = {
+    case j: JsdPfVat => j.chargeRef
     case _           => throw new RuntimeException("Incorrect origin found")
   }
 
+  override def checkYourAnswersAdditionalReferenceRow(journeyRequest: JourneyRequest[AnyContent])(payFrontendBaseUrl: String): Option[CheckYourAnswersRow] = {
+    chargeReference(journeyRequest.journey.journeySpecificData).map { chargeReference =>
+      CheckYourAnswersRow(
+        titleMessageKey = "check-your-details.PfVat.charge-reference",
+        value           = Seq(chargeReference.value),
+        changeLink      = None
+      )
+    }
+  }
+
+  override def openBankingOriginSpecificSessionData: JourneySpecificData => Option[OriginSpecificSessionData] = {
+    case j: JsdPfVat => for {
+      vrn <- j.vrn
+      vatChargeRef <- j.chargeRef
+    } yield PfVatSessionData(vrn, Some(vatChargeRef))
```

Comment:
should this not be the optional charge ref? instead of `Some(vatChargeRef)` just make it j.chargeRef